src/content/docs/en/recipes/dynamically-importing-images.mdx

```diff
@@ -0,0 +1,132 @@
+---
+title: Dynamically Import Images
+description: Learn how to dynamically import images using Vite's import.meta.glob function
+i18nReady: true
+type: recipe
+---
+import FileTree from '~/components/FileTree.astro'
+
+When working with custom components that include images, there will be times where you will need to dynamically import the image paths. 
+
+In this recipe, you will learn how to dynamically import your images the correct way using Vite's `import.meta.glob` function. 
+
+## Recipe
+
+1. Create a new `assets` folder under the `src` directory and add your images inside that new folder. 
+
+<FileTree>
+- src/
+  - assets/
+    - example-image.jpg
+    - example-image-2.jpg
+    - example-image-3.jpg
+</FileTree>
+
+:::note
+`assets` is a popular folder name convention for placing images but you are free to name the folder whatever you like.
+::: 
+
+2. Create a new Astro component and import the `<Image>` component.
+
+    ```astro title="src/components/MyCustomCardComponent.astro" 
+    ---
+     import { Image } from 'astro:assets';
+    ---
+    ```
+
+3. Create the `props` for your component. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={4-11}
+    ---
+     import { Image } from 'astro:assets';
+
+     interface Props {
+        imagePath: string;
+        altText: string;
+        name: string;
+        age: number;
+     }
+
+     const { imagePath, altText, name, age } = Astro.props;
+    ---
+    ``` 
+
+4. Create a new `images` variable and use the `import.meta.glob` function which returns an object of all of the image paths inside the `assets` folder. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={12}
+    ---
+     import { Image } from 'astro:assets';
+
+     interface Props {
+        imagePath: string;
+        altText: string;
+        name: string;
+        age: number;
+     }
+     
+     const { imagePath, altText, name, age } = Astro.props;
+     const images = import.meta.glob<Record<string, () => Promise<{ default: ImageMetadata }>>('/src/**/*.{jpeg,jpg,png,gif}')
```

Comment:
@Princesseuh it's not required to provide the generic right? That's the default signature. Maybe we could just add it in a comment above or below?

Reply:
Agreed with @florian-lefebvre here: I'd love to see that original import function so as not to startle our non-TypeScript users.

Similar to my comment above about defining the prop types as optional, if it's helpful for TS users to know the shape, then after this code snippet there could be a mention separately of it.